### PR TITLE
feat: api return schedule

### PR DESCRIPTION
**Pull request by emrysal**
## What does this PR do?

* Implements scheduleId to apps/api/pages/api/event-types/[id]/_get.ts and apps/api/pages/api/event-types/_get.ts
* Some style fixes
* Some guard fixes

### CODE REVIEW

File: apps/api/pages/api/event-types/[id]/_get.ts [hunk 1]
```diff
@@ -51,10 +54,43 @@ export async function getHandler(req: NextApiRequest) {
       owner: { select: { username: true, id: true } },
     },
   });
+  checkPermissions(req, eventType);
```
Comment (zomars) [inline apps/api/pages/api/event-types/[id]/_get.ts:57]:
```suggestion
  await checkPermissions(req, eventType);
```

File: apps/api/pages/api/event-types/[id]/_get.ts [hunk 2]
```diff
@@ -51,10 +54,43 @@ export async function getHandler(req: NextApiRequest) {
...
+}
 
-  const link = event_type ? getCalLink(event_type) : null;
+type BaseEventTypeCheckPermissions = {
+  userId: number | null;
+  teamId: number | null;
+};
 
-  return { event_type: schemaEventTypeReadPublic.parse({ ...event_type, link }) };
+function checkPermissions<T extends BaseEventTypeCheckPermissions>(
```
Comment (zomars) [inline apps/api/pages/api/event-types/[id]/_get.ts:83]:
```suggestion
async function checkPermissions<T extends BaseEventTypeCheckPermissions>(
```

File: apps/api/pages/api/event-types/[id]/_get.ts [hunk 3]
```diff
@@ -51,10 +54,43 @@ export async function getHandler(req: NextApiRequest) {
...
+  req: NextApiRequest,
+  eventType: (T & Partial<Omit<T, keyof BaseEventTypeCheckPermissions>>) | null
+) {
+  if (req.isAdmin) return true;
+  if (eventType?.teamId) {
+    req.query.teamId = String(eventType.teamId);
+    canAccessTeamEventOrThrow(req, "MEMBER");
```
Comment (zomars) [inline apps/api/pages/api/event-types/[id]/_get.ts:90]:
```suggestion
    await canAccessTeamEventOrThrow(req, "MEMBER");
```

**Review by zomars — CHANGES_REQUESTED** [review]
We need to await permissions checks before continuing

**Review by zomars — APPROVED** [review]
Ship it

**Comment by emrysal** [comment]
@zomars Excellent spot! missed that 🙏

**Pull request merged**